// @jser/stat/lib/models/JSerItem.js
// LICENSE : MIT
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JSerItem = void 0;
const assert = require("assert");
const JSerItemRelatedLink_1 = require("./JSerItemRelatedLink");
class JSerItem {
    constructor(item) {
        /** @type {string} */
        this.title = item["title"];
        /** @type {string} */
        this.url = item["url"];
        /** @type {string} */
        this.content = item["content"];
        /** @type {string[]} */
        this.tags = item["tags"] || [];
        /** @type {Date} */
        this.date = new Date(item["date"]);
        var relatedLinks = item["relatedLinks"] || [];
        /** @type {JSerItemRelatedLink[]} */
        this.relatedLinks = relatedLinks.map(function (link) {
            return new JSerItemRelatedLink_1.default(link);
        });
    }
    /**
     * @param {JSerItem} item
     * @returns {boolean}
     */
    isEqualItem(item) {
        assert(item != null, "item should not be null");
        return this.url === item.url;
    }
}
exports.JSerItem = JSerItem;
